### PR TITLE
feat: use rich to manage ui

### PR DESCRIPTION
**Pull request by samsja**
Use rich to handle UI

Action items for this ticket includes:
- [x] refactor progressbar UI to use `rich`
- [x] refactor all spinner/status UI to use `rich`
- [x] define logging with rich handler
- [x] delete old code on CLI UI

- [x] what about rich traceback ? out of the scope of this PR

- [x] fix pycharm bug :  see https://github.com/Textualize/rich/issues/206

issue : https://github.com/jina-ai/jina/issues/4395

new ProgressBar with rich :

richlogging

old one

master

I tried to keep it as close as how it was before

### CODE REVIEW

File: jina/logging/profile.py [hunk 1]
```diff
@@ -6,13 +6,121 @@
...
+            ),
+        ]
+
+        super().__init__(*columns, **kwargs)
+
+        self.task_id = self.add_task("Working...", total=total_length)
+
+    @overload
+    def update(self):
+        """Update the progress bar"""
```
Comment (Okayyyhokay12) [inline jina/logging/profile.py:73]: Get the duck off my phone bitch

**Review by Okayyyhokay12 — APPROVED** [review]

File: jina/logging/profile.py [hunk 2]
```diff
@@ -6,13 +6,121 @@
...
+        ...
+
+    @overload
+    def update(self, task_id: TaskID, *args, **kwargs):
```
Comment (JoanFM) [inline jina/logging/profile.py:77]: Why to overload this? the one below also includes this right?

File: jina/logging/profile.py [hunk 3]
```diff
@@ -6,13 +6,121 @@
...
+            TextColumn('ETA:', style='progress.remaining'),
+            TimeRemainingColumn(),
+            _OnDoneColumn(
+                '{task.completed} steps done in {task.finished_time:.0f} seconds'
+            ),
+        ]
+
+        super().__init__(*columns, **kwargs)
+
+        self.task_id = self.add_task("Working...", total=total_length)
```
Comment (JoanFM) [inline jina/logging/profile.py:69]: please use single quotes

File: jina/logging/profile.py [hunk 4]
```diff
@@ -6,13 +6,121 @@
...
+            with ProgressBar('loop'):
+                do_busy()
+    """
+
+    def __init__(
+        self,
+        total_length: int,
+        description: str = 'Working...',
+        columns: Optional[Union[str, ProgressColumn]] = None,
+        message_on_done: Union[str, Callable[..., str], None] = None,  # todo implement
```
Comment (JoanFM) [inline jina/logging/profile.py:49]: Please do not use `Union[..., None]` but use `Optional`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/resources/extra-requirements.txt [hunk 5]
```diff
@@ -72,3 +72,11 @@ portforward>=0.2.4:         cicd
 kubernetes>=18.20.0:        perf, standard, devel
 pytest-kind==21.1.3:        test
 pytest-lazy-fixture:        test
+datasets:                   cicd
```
Comment (JoanFM) [inline jina/resources/extra-requirements.txt:75]: thse were removed, u can remove them

File: extra-requirements.txt [hunk 6]
```diff
@@ -72,3 +72,4 @@ portforward>=0.2.4:         cicd
 kubernetes>=18.20.0:        perf, standard, devel
 pytest-kind==21.1.3:        test
 pytest-lazy-fixture:        test
+rich:                       core
```
Comment (hanxiao) [inline extra-requirements.txt:75]: no need, as docarray includes it already

File: jina/resources/logging.default.yml [hunk 7]
```diff
@@ -1,5 +1,5 @@
 handlers:  # enabled handlers, order does not matter
-  - StreamHandler
```
Comment (JoanFM) [inline jina/resources/logging.default.yml:2]: what is this change, what does it imply? 

Can u add both Handlers as default?

Comment (samsja) [reply jina/resources/logging.default.yml:2]: The change is now that the main logger is the Rich logger. If we had both, the log are going to be print two times, one with rich one with the StreamHandler

File: jina/orchestrate/flow/base.py [hunk 8]
```diff
@@ -1139,38 +1137,40 @@ def _polling_status():
             threads.append(t)
             t.start()
 
-        # kick off spinner thread
-        t_m = threading.Thread(target=_polling_status, daemon=True)
-        t_m.start()
+        console = Console()
+        with console.status('Wokring...') as status:
```
Comment (JoanFM) [inline jina/orchestrate/flow/base.py:1141]:
```suggestion
        with console.status('Working...') as status:
```

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/orchestrate/flow/base.py (hunk 8 above)
Comment (samsja) [reply jina/orchestrate/flow/base.py:1141]: It looks exactly the same for me haha

File: jina/helloworld/fashion/helper.py [hunk 9]
```diff
@@ -187,13 +187,17 @@ def download_data(targets, download_proxy=None, task_name='download fashion-mnis
             {'http': download_proxy, 'https': download_proxy}
         )
         opener.add_handler(proxy)
+    from rich import print
```
Comment (JoanFM) [inline jina/helloworld/fashion/helper.py:190]: this import does not seem needed

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/resources/logging.default.yml [hunk 10]
```diff
@@ -15,3 +15,9 @@ configs:
     host: # when not given then record it locally, /dev/log on linux /var/run/syslog on mac
     port: # when not given then record it locally,  /dev/log on linux /var/run/syslog on mac
     formatter: PlainFormatter
+  RichHandler:
```
Comment (JoanFM) [inline jina/resources/logging.default.yml:18]: I am not sure if the RichHandler could be causing problems

**Review by JoanFM — CHANGES_REQUESTED** [review]
When I run a `Flow` the `protocol` shows a number, there should be some option to display Enums as `strings`

**Comment by JoanFM** [comment]
The test is blocked after success, something is not properly handled with multiprocessing and I'd bet on the RichHandler.

Also, when putting Logs in debug mode the coloring seems very weird

File: jina/resources/logging.default.yml (hunk 10 above)
Comment (samsja) [reply jina/resources/logging.default.yml:18]: The RichHandler is here but I don't use it anymore as it breaks logging in docker and k8s

**Review by JoanFM — APPROVED** [review]

**Pull request merged**